### PR TITLE
fix(controller): retry transient error on agent pod creation

### LINKED ISSUE #13654: When creating an agent pod, transient error is not retried
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I have tested with the `:latest` image tag (i.e. `quay.io/argoproj/workflow-controller:latest`) and can confirm the issue still exists on `:latest`. If not, I have explained why, **in detail**, in my description below.
- [X] I have searched existing issues and could not find a match for this bug
- [X] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/main/docs/CONTRIBUTING.md))

## What happened ？
When I use the HTTP template, I find it pending all the time. 
image
Then I saw that the workflow status was error, and encountered a transient error.
image
error is : `failed to create Agent pod. Reason: Operation cannot be fulfilled on resourcequotas "xxxx": the object has been modified; please apply your changes to the latest version and try again`

## What did you expect to happen?

I hope to encounter transient errors that can be retried.

### Version(s)

v3.4.17

### Paste a minimal workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
None
```

### Logs from the workflow controller

```text
None
```

### Logs from in your workflow's wait container

```text
None
```

### PR DESCRIPTION
**Pull request by fyp711**
Fixes #13654 

### Motivation

When I use the HTTP template, I find it pending all the time. 
image
Then I saw that the workflow status was error, and encountered a transient error.
image
error is : `failed to create Agent pod. Reason: Operation cannot be fulfilled on resourcequotas "xxxx": the object has been modified; please apply your changes to the latest version and try again`

### Modifications
When the agent pod creation encounters transient errors, just retry.

### CODE REVIEW

**Comment by fyp711** [comment]
Could anyone help to rerun it ?
image

**Comment by MasonM** [comment]
@fyp711 That errors should hopefully be fixed with https://github.com/argoproj/argo-workflows/pull/13660. After that's merged, click the "Update branch" button at the bottom of the PR

**Comment by fyp711** [comment]
> @fyp711 That error should hopefully be fixed with #13660. After that's merged, click the "Update branch" button at the bottom of the PR

Updated thanks.

**Comment by fyp711** [comment]
@agilgur5 @Joibel Could you help to review this pr ? If you have free time thanks.

**Review by shuangkun — APPROVED** [review]
LGTM

File: workflow/controller/agent.go [hunk 1]
```diff
@@ -257,6 +258,10 @@ func (woc *wfOperationCtx) createAgentPod(ctx context.Context) (*apiv1.Pod, erro
 				return existing, nil
 			}
 		}
+		if errorsutil.IsTransientErr(err) {
+			woc.requeue()
+			return created, nil
```
Comment (juliev0) [inline workflow/controller/agent.go:263]: I would think that `created` would be nil if we get here, right? Or am I wrong?

Comment (juliev0) [reply workflow/controller/agent.go:263]: Okay, never mind - I see you're just being complete, in the case that some transient error does result in a non-nil Pod for some reason.

**Review by juliev0 — APPROVED** [review]

**Pull request merged**